mesheryctl/internal/cli/root/system/model.go

```diff
@@ -413,6 +425,133 @@ mesheryctl exp model search [query-text]
+	Example: `
+
+// Import a model
+mesheryctl exp model import --file <URI>
+	`,
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp model import <URI/path to Model>\nRun 'mesheryctl exp model import --help' to see detailed help message"
+		if len(args) == 0 {
+			return fmt.Errorf("Path/URI term is missing\n\n%v", errMsg)
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Here we have to call the model import API /api/meshmodel/components/register with POST
+		// The body of the request will be the json of type below
+		// type MeshModelRegistrantData struct {
+		// 	Host       Host                 `json:"host"`
+		// 	EntityType types.CapabilityType `json:"entityType"`
+		// 	Entity     []byte               `json:"entity"` //This will be type converted to appropriate entity on server based on passed entity type
+		// }
+
+		// Check if file path is provided
+		if validURL := govalidator.IsURL(args[0]); !validURL {
+			if _, err := os.Stat(args[0]); os.IsNotExist(err) {
+				return errors.New("file path does not exist")
+			}
+			// get the directory path
+			dirPath := args[0]
+			// read the files in the directory
+			err := processDirectory(dirPath)
+			if err != nil {
+				fmt.Println("Error processing directory:", err)
+				return err
+			}
+		} else if isGithubURL := strings.Contains(args[0], "github.com"); isGithubURL {
+			// Here we will be reading the content of the file from the URL
+
+			// Parse the URL
+			parsedURL, err := url.Parse(args[0])
+			if err != nil {
+				fmt.Println("Error parsing URL:", err)
+				return err
+			}
+
+			// Extract parts
+			pathParts := strings.Split(parsedURL.Path, "/")
+
+			githubInfo := walker.NewGithub()
+			githubInfo.Owner(pathParts[1])
+			githubInfo.Repo(pathParts[2])
+			githubInfo.Branch(pathParts[4])
+			githubInfo.Root(strings.Join(pathParts[5:], "/"))
+			FileInterceptor := func(file walker.GithubContentAPI) error {
+
+				// Download the file content
+				contentResponse, err := http.Get(file.DownloadURL)
+				if err != nil {
+					return err
+				}
+				defer contentResponse.Body.Close()
+
+				// Read the file content directly into a []byte
+				fileContent, err := ioutil.ReadAll(contentResponse.Body)
+				if err != nil {
+					return err
+				}
+
+				// Register the file content as a component
+				if err := registerComponent(file.Name, fileContent); err != nil {
+					return err
+				}
+
+				return nil
+			}
+
+			githubInfo.RegisterFileInterceptor(FileInterceptor)
+
+			if err := githubInfo.Walk(); err != nil {
+				fmt.Println("Error occurred during traversal:", err)
+				return err
+			}
+		} else {
+			// If the user wants to import the model from the oci registry
+
+			// Here extract the registry address and tag
+			// docker.io/repo/container:tag
+			reg := strings.Split(args[0], ":")
+			regAddress := reg[0]
+			tag := reg[1]
+			// 0. Create an OCI layout store
+			fs, err := file.New("./tmp/")
```

Comment:
Do not specify this path as `tmp`. 

One of two things should be done. Either:

1. as the OS for a location (use golang library)
1. use mesheryctl's designated location of `~/.meshery/.somwhere./.in../.here./...`